### PR TITLE
[pytorch] [trcomp] [huggingface] add trcomp 1.9 dockerfile with latest CVE fixes

### PR DESCRIPTION
**Pull request by harishneit**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: huggingface/pytorch/buildspec-trcomp-1-9.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
+region: &REGION <set-$REGION-in-environment>
+base_framework: &BASE_FRAMEWORK pytorch
+framework: &FRAMEWORK !join [ "huggingface_", *BASE_FRAMEWORK, "_trcomp"]
+version: &VERSION 1.9.0
```
Comment (Lokiiiiii) [inline huggingface/pytorch/buildspec-trcomp-1-9.yml:5]: HF is currently at 1.9.1
https://github.com/aws/deep-learning-containers/blob/master/available_images.md

File: huggingface/pytorch/buildspec-trcomp-1-9.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  BuildHuggingFacePytorchTrCompGpuPy38Cu111TrainingDockerImage:
+    <<: *TRAINING_REPOSITORY
+    build: &HUGGINGFACE_PYTORCH_TRCOMP_GPU_TRAINING_PY3 false
+    image_size_baseline: &IMAGE_SIZE_BASELINE 18000
+    device_type: &DEVICE_TYPE gpu
+    python_version: &DOCKER_PYTHON_VERSION py3
+    tag_python_version: &TAG_PYTHON_VERSION py38
+    cuda_version: &CUDA_VERSION cu111
+    os_version: &OS_VERSION ubuntu20.04
+    transformers_version: &TRANSFORMERS_VERSION 4.11.0
```
Comment (Lokiiiiii) [inline huggingface/pytorch/buildspec-trcomp-1-9.yml:26]: HF-PT 1.9.1 was released with transformers 4.12.3

File: huggingface/pytorch/training/docker/1.9/py3/cu111/Dockerfile.trcomp.gpu [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# https://github.com/aws/deep-learning-containers/blob/master/available_images.md 
+# refer to the above page to pull latest Pytorch image
+
+# docker image region us-west-2
+FROM 763104351884.dkr.ecr.us-west-2.amazonaws.com/pytorch-training:1.9-gpu-py38-cu111-ubuntu20.04-v1
```
Comment (Lokiiiiii) [inline huggingface/pytorch/training/docker/1.9/py3/cu111/Dockerfile.trcomp.gpu:5]: I believe this image is 1.9.1 as well. That was the latest Base DLC release.

**Review by Lokiiiiii — COMMENTED** [review]
Looks like HF moved from [PT-1.9.0-HF-4.11.0 to PT-1.9.1-HF-4.12.3](https://github.com/aws/deep-learning-containers/pull/1607) and we missed this. Need to align on whether we need to update these versions now.

I advise against doing a new release for PT 1.9 but let's do our due diligence and capture approvals here.

File: huggingface/pytorch/training/docker/1.9/py3/cu111/Dockerfile.trcomp.gpu (hunk 3 above)
Comment (arjkesh) [reply huggingface/pytorch/training/docker/1.9/py3/cu111/Dockerfile.trcomp.gpu:5]: I believe that you are uninstalling the underlying binary from this image and re-installing a new one right? Thus the version needs to correspond to the pytorch version that is ultimately installed in the container

File: dlc_developer_config.toml [hunk 4]
```diff
@@ -15,7 +15,7 @@ graviton_mode = false
 habana_mode = false
```
Comment (arjkesh) [inline dlc_developer_config.toml:15]: Please revert the config file and we will merge

**Comment by harishneit** [comment]
@arjkesh I rebased with `master` and reverted `dlc_developer_config.toml` changes

**Review by Lokiiiiii — APPROVED** [review]

**Review by arjkesh — APPROVED** [review]

**Pull request merged**